src/features/groups/components/MemberList.tsx

```diff
@@ -1,21 +1,29 @@
-import { IconPersonCircle, IconPlus } from '@/assets/images/icons';
-import { Button, Icon } from '@/components/ui';
+import { IconPersonCircle } from '@/assets/images/icons';
+import { Icon } from '@/components/ui';
 import { fetchUserList } from '@/lib/api/user/server';
-import { IUser } from '@/types/definition';
+import { IGroup, IUser } from '@/types/definition';
 
 import Link from 'next/link';
 
-export const MemberList = async ({ id }: { id: string }) => {
-  const users: IUser[] = await fetchUserList(id);
+import { InviteMemberDialog } from './InviteMemberDialog';
+
+interface IMemberListProps {
+  /**
+   * an identifier of a group which members belong to
+   */
+  groupId: IGroup['id'];
+}
+
+export const MemberList = async ({ groupId }: IMemberListProps) => {
+  const users: IUser[] = await fetchUserList(groupId);
+
   return (
     <div className="mb-6">
       <div className="flex items-center justify-between mb-2 h-12">
         <h2 className="text-xl">Members</h2>
-        <Button className="flex justify-center items-center w-12 h-12">
-          <Icon icon={IconPlus} size={4.5} />
-        </Button>
+        <InviteMemberDialog groupId={groupId} />
```

Comment:
The two plus buttons on the members page and the group single page are too dependent on each other.

They should be independent, while you can reuse the dialog content.

Reply:
@nick-y-ito Thank you for review and advice here. let me confirm with you that I can reuse function inclusing dialog. but the trigger should be indepenedent for high flexibility and scalability ( in another word make it loose coupling) ?

Reply:
@nick-y-ito Thank you ! I created new component named `InviteMemberDialogTrigger` which is a trigger of invitaiton dialog.